control-plane/pkg/reconciler/consumer/consumer.go

```diff
@@ -307,16 +307,16 @@ func addResource(resource *contract.Resource) contractMutatorFunc {
 func removeResource(_ *zap.Logger, ct *contract.Contract, c *kafkainternals.Consumer) int {
 	idx := coreconfig.FindResource(ct, c.GetUID())
 	if idx == coreconfig.NoResource {
-		return coreconfig.NoResource
+		return coreconfig.ResourceUnchanged
 	}
 	coreconfig.DeleteResource(ct, idx)
-	return idx
+	return coreconfig.ResourceChanged
```

Comment:
I believe this was the reason.

when we use this function we check for `changed == coreconfig.ResourceChanged` to then proceed to update the CM, however, by returning `idx`, `changed == coreconfig.ResourceChanged` is almost never true, so we never update the CM with the resource removed.

Reply:
seems reasonable, perhaps to add a comment?

Thanks for the reasoning here

Reply:
FWIW, we can add a comment for the upstream PR... just saying (if we want to ...)